### ISSUE TITLE
Helm install fails with 'incompatible types for comparison' when using Helm v2.15.0(-rc)

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#23516]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Use latest RC for Helm (v2.15.0-rc2) and try to install Rancher

**Result:**
```
Error: render error in "rancher/templates/deployment.yaml": template: rancher/templates/deployment.yaml:76:7: executing "rancher/templates/deployment.yaml" at <gt .Values.auditLog.level 0.0>: error calling gt: incompatible types for comparison
```

**Other details that may be helpful:**
Helm merged https://github.com/helm/helm/pull/6010 where they move away from float64 when parsing numbers and now `json.Number` is returned.

Workarounds that have been applied in other cases (parsing to `int`) also don't work as anything parsed will just be `0`.

What seems to work is `{{- if gt (.Values.auditLog.level | toString | atoi) 0 }}` but this needs some additional testing

**Comment by mablanco** [comment]
Hi @superseb I'm having the same issue, but when upgrading Rancher in one of the cluster I maintain. I'd gladly help you testing for a workaround (or a solution). Just let me know. Best regards.

**Comment by superseb** [comment]
A fix for backwards compatibility will be made via https://github.com/helm/helm/issues/6708 in v2.15.1. Using the latest v2.14 should unblock you for now.

**Comment by tmbs** [comment]
Same error with Helm v2.14.3, tried upgrading from v2.2.8 to v2.2.9:
```
$ helm version
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
```
```
$ helm upgrade rancher rancher-stable/rancher \
> --set=hostname=*** \
> --set=auditLog.level=auditLog.level.3 \
> --set=ingress.tls.source=secret
UPGRADE FAILED
Error: render error in "rancher/templates/deployment.yaml": template: rancher/templates/deployment.yaml:76:7: executing "rancher/templates/deployment.yaml" at <gt .Values.auditLog.level 0.0>: error calling gt: incompatible types for comparison
Error: UPGRADE FAILED: render error in "rancher/templates/deployment.yaml": template: rancher/templates/deployment.yaml:76:7: executing "rancher/templates/deployment.yaml" at <gt .Values.auditLog.level 0.0>: error calling gt: incompatible types for comparison
```

**Comment by alena1108** [comment]
https://github.com/helm/helm/issues/6708

**Comment by icanhazbroccoli** [comment]
@alena1108 helm 2.14.x doesn't contain this patch, therefore the issue might be caused by a different reason.

**Comment by rpd5003** [comment]
not sure if this is the proper fix....
1. Updated deployment.yml to remove `replicas := int (.Values.replicas)`. This was causing a YAML to JSON mapping error. replicas did not seem to be referenced, so removed this line
2. Several lines in deployment.yml had a comparison like `{{- if gt .Values.auditLog.level 0.0 }}` which caused the incompatible types issue. Changed all lines to `{{- if gt (float64 (toString (.Values.auditLog.level))) 0.0 }}`

These steps seemed to allow the upgrade to complete successfully....

**Comment by rpd5003** [comment]
actually I think item #1 above can be ignored after the fix to https://github.com/helm/helm/issues/6708

That fixed the replicas YAML to JSON issue, but did not resolve the "gt" comparison issue

**Comment by rpd5003** [comment]
> A fix for backwards compatibility will be made via [helm/helm#6708](https://github.com/helm/helm/issues/6708) in v2.15.1. Using the latest v2.14 should unblock you for now.

@icanhazbroccoli I see the fix has been merged into the dev-v2 branch. Do you know when that will be part of a formal release?

@superseb, I assume we use the workaround until the formal release?

**Comment by icanhazbroccoli** [comment]
@rpd5003 I don't have this data, sorry. Feel free to reach out core team folks int helm-dev slack channel to check this.

**Comment by superseb** [comment]
Helm v2.15.1 (https://github.com/helm/helm/releases/tag/v2.15.1) resolves this issue.

**Comment by mablanco** [comment]
Hi, I've just tried Helm v2.15.1 and I've been able to update my test Rancher cluster from 2.2.3 to 2.3.0 without problems. It seems that helm works again.

**Comment by SvenC56** [comment]
I have still the same error. Looking at https://releases.rancher.com/server-charts/latest you can see the following error message: `The specified key does not exist`

**Comment by icanhazbroccoli** [comment]
@SvenC56 it looks like something else: I don't see anything yelling numeric regression here. @superseb is it something you guys could check please?

**Comment by superseb** [comment]
@SvenC56 Please show the exact commands used from https://rancher.com/docs/rancher/v2.x/en/installation/ha/ and the output. The file used by Helm is https://releases.rancher.com/server-charts/latest/index.yaml and `curl https://releases.rancher.com/server-charts/latest/index.yaml` should work fine.

**Comment by SvenC56** [comment]
I will provide you the output on monday. I followed the HA [installation guide of rancher](https://rancher.com/docs/rancher/v2.x/en/installation/ha/). These are the steps i followed:

```
curl https://get.helm.sh/helm-v2.15.1-linux-amd64.tar.gz

tar -zxvf helm-v2.15.1-linux-amd64.tar.gz

chmod +x linux-amd64/helm

mv linux-amd64/helm /usr/local/bin/helm

...

kubectl -n kube-system create serviceaccount tiller

kubectl create clusterrolebinding tiller \
  --clusterrole=cluster-admin \
  --serviceaccount=kube-system:tiller

helm init --service-account tiller

kubectl -n kube-system  rollout status deploy/tiller-deploy

helm repo add rancher-stable https://releases.rancher.com/server-charts/stable

**Comment by superseb** [comment]
any update on this @SvenC56 ?

**Comment by SvenC56** [comment]
Hi,

in conclusion it was a bit of a different error. We are behind a corperate network. We found the issue in the CA Trust chain. After we added the CA Certs I could download the charts without a problem.

**Comment by superseb** [comment]
Ok thanks!

We have added documentation for Helm versions here: https://rancher.com/docs/rancher/v2.x/en/installation/helm-version/

**Comment by wzrdtales** [comment]
this seems to be still broken on helm v3 @superseb tried it on helm v3.0.2. however the latest docs suggest to use helm v3 not v2

**Comment by naveen230** [comment]
@superseb still having the issue when trying to upgrade rancher with new ssl certificates
"helm get values rancher" returns
Error: release: "rancher" not found

helm repo list returns
NAME          	URL                                              
stable        	https://kubernetes-charts.storage.googleapis.com 
local         	http://127.0.0.1:8879/charts                     
rancher-stable	https://releases.rancher.com/server-charts/stable
jetstack      	https://charts.jetstack.io   

helm version
Client: &version.Version{SemVer:"v2.16.2", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.2", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
